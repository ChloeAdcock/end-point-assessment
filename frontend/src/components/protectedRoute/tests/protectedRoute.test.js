import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import ProtectedRoute from "../ProtectedRoute";

const FakeComponent = () => {
  return <p>Fake component</p>;
};

describe("The protected route component", () => {
  it("should render the component when a user is authenticated", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ProtectedRoute component={FakeComponent} user="fakeuser" />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
