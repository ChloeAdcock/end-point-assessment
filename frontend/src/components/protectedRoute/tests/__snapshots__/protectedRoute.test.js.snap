// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The protected route component should render the component when a user is authenticated 1`] = `
<p>
  Fake component
</p>
`;
